pinot-spi/src/main/java/org/apache/pinot/spi/utils/NetUtils.java

```diff
@@ -57,4 +59,29 @@ public static String getHostnameOrAddress() {
       }
     }
   }
+
+  /**
+   * Find an open port.
+   * @return an open port
+   * @throws IOException
+   */
+  public static int findOpenPort()
+      throws IOException {
+    try (ServerSocket socket = new ServerSocket(0)) {
+      return socket.getLocalPort();
+    }
+  }
+
+  /**
+   * Find an open port，otherwise use given default port.
+   * @param defaultPort
+   * @return an open port otherwise default port
+   */
+  public static int findOpenPort(int defaultPort) {
```

Comment:
Can we first try with the default port, and use open port only when default is occupied? This can save the step of looking for port when using integration test to debug stuff.

Reply:
Also, it might be better to try `defaultPort + i` until we get an open port so that the port number is more deterministic

Reply:
done